src/frontend/src/components/Shape/ShapeCreate.js

```diff
@@ -28,7 +28,7 @@ const ShapeCreate = props => {
 
     document.addEventListener("click", documentClickHandler)
     return () => document.removeEventListener("click", documentClickHandler)
-  }, [ghostPoint])
+  }, [ghostPoint, shape, transformation])
```

Comment:
@tajpouria does it change the renders amount to add shape & transformation to useEffect here?

Reply:
ShapCreate.js contains linting warnings, you can fix it either by adding it's dependencies as second arguments or put // eslint-disable-next-line react-hooks/exhaustive-deps before the violating line if there is a good reason. Usually disabling it is a mistake and will significantly bite you later. (Most cases where people think it's safe to disable are not safe and lead to stale closures.)

Some references for how to write code with correctly declared dependencies:

https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies
https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often
Concrete cases where people thought it's safe but it wasn't, with correct solutions: facebook/react#14920 (comment)

Reply:
Here's another [PR #29](https://github.com/ninabondar/epigma/pull/29) without changing the mentioned hooks and manifest.json.